homeassistant/components/solarlog/sensor.py

```diff
@@ -41,14 +42,5 @@ def __init__(
     @property
     def native_value(self):
         """Return the native sensor value."""
-        if self.entity_description.key == "time":
-            state = as_local(
-                getattr(self.coordinator.data, self.entity_description.key)
-            )
-        else:
-            result = getattr(self.coordinator.data, self.entity_description.key)
-            if self.entity_description.factor:
-                state = round(result * self.entity_description.factor, 3)
-            else:
-                state = result
-        return state
+        state = getattr(self.coordinator.data, self.entity_description.key)
+        return cast(StateType, self.entity_description.value(state))
```

Comment:
Is the cast necessary here?
You need to be careful as `StateType` doesn't include `datetime`, and I think `as_local` returns a `datetime`.

Reply:
I was struggling a bit with the datetime value from one of the sensors, yes. 

But you're right, we can remove the cast.